docs/api-reference/sql-api.md

```diff
@@ -23,167 +23,287 @@ sidebar_label: Druid SQL
-
-You can use _curl_ to send SQL queries from the command-line:
-
-```bash
-$ cat query.json
-{"query":"SELECT COUNT(*) AS TheCount FROM data_source"}
-
-$ curl -XPOST -H'Content-Type: application/json' http://ROUTER:8888/druid/v2/sql/ -d @query.json
-[{"TheCount":24433}]
-```
+This endpoint also supports querying metadata by using [metadata tables](../querying/sql-metadata-tables.md).
+
+#### URL
+
+<code class="postAPI">POST</code> <code>/druid/v2/sql</code>
+
+#### Request body
+
+* `query`: SQL query string.
+* `resultFormat`: Format of query results.
+  * `object`: Returns a JSON array of JSON objects with the HTTP header `Content-Type: application/json`.
+  * `array`: Returns a JSON array of JSON arrays with the HTTP header `Content-Type: application/json`.
+  * `objectLines`: Returns newline-delimited JSON objects with a trailing blank line. Sent with the HTTP header `Content-Type: text/plain`.
+  * `arrayLines`: Returns newline-delimited JSON arrays with a trailing blank line. Sent with the HTTP header `Content-Type: text/plain`. 
+  * `csv`: Returns a comma-separated values with one row per line and a trailing blank line. Sent with the HTTP header `Content-Type: text/csv`. 
+* `header`: Adds a header row with information on column names for the query result when set to `true`. You can optionally include `typesHeader` and `sqlTypesHeader`. 
+
+    Complex types, like sketches, will be reported as `COMPLEX<typeName>` if a particular complex type name is known for that field, or as `COMPLEX` if the particular type name is unknown or mixed. 
+    
+    If working with an older version of Druid, set `header` to `true` to verify compatibility. Druid returns the HTTP header `X-Druid-SQL-Header-Included: yes` when the client connects to an older Druid version with support for the `typesHeader` and `sqlTypesHeader` parameters. Additionally, Druid returns a `X-Druid-SQL-Query-Id` HTTP header with the value of `sqlQueryId` from the [query context parameters](../querying/sql-query-context.md) if specified, else Druid will generate a SQL query ID.
+* `typesHeader`: Adds Druid runtime type information in the header. Can only be set when `header` is also `true`.
+* `sqlTypesHeader`: Adds SQL type information in the header. Can only be set when `header` is also `true`.
+* `context`: JSON object containing optional [SQL query context parameters](../querying/sql-query-context.md), such as to set the query ID, time zone, and whether to use an approximation algorithm for distinct count.
+* `parameters`: List of query parameters for parameterized queries. Each parameter in the list should be a JSON object like `{"type": "VARCHAR", "value": "foo"}`. The type should be a SQL type; see [Data types](../querying/sql-data-types.md) for a list of supported SQL types.
+
```

Comment:
Making the example more prominent with the property name will be easier to follow. Another option that could help users is to include the parameterized query as a separate example, similar to what was in the docs
![image](https://github.com/apache/druid/assets/7747997/9e952721-637c-4f44-9adb-9aadfd94c1ab)


`````suggestion
For example:
   ```
    "parameters": [
        {
            "type": "TIMESTAMP",
            "value": "2019-08-21 00:00:00"
        }
    ]
   ```
`````